src/fragments/products.ts

```diff
@@ -224,8 +224,41 @@ export const productFragmentDetails = gql`
   }
 `;
 
-export const fragmentVariant = gql`
+export const selectedProductTypeVariantAttributeFragment = gql`
+  ${attributeValueFragment}
+  fragment SelectedProductTypeVariantAttributeFragment on Attribute {
+    id
+    name
+    slug
+    inputType
+    valueRequired
+    values {
+      ...AttributeValueFragment
+    }
+  }
+`;
+
+export const selectedProductVariantAttributeFragment = gql`
   ${attributeValueFragment}
+  fragment SelectedProductVariantAttributeFragment on SelectedAttribute {
+    attribute {
+      id
+      name
+      slug
+      inputType
+      valueRequired
+      values {
+        ...AttributeValueFragment
+      }
+    }
+    values {
+      ...AttributeValueFragment
+    }
+  }
+`;
+
```

Comment:
Attributes can be also selected on variants, and I believe those have the same schema (correct me if im wrong)
```suggestion
export const attributeFragment = gql`
  ${attributeValueFragment}
  fragment AttributeFragment on Attribute {
    id
    name
    slug
    inputType
    valueRequired
    values {
      ...AttributeValueFragment
    }
  }
`;

export const selectedAttributeFragment = gql`
  ${attributeValueFragment}
  ${attributeFragment}
  fragment SelectedProductVariantAttributeFragment on SelectedAttribute {
    attribute {
      ...AttributeFragment
    }
    values {
      ...AttributeValueFragment
    }
  }
`;

```

Reply:
Yep, I'll change it, although I'll give it other name e.g. VariantAttributeFragment, as AttributeFragment is already in scope and this attribute fragment is intended to use in variants.

Reply:
Product and variant attributes are both type of attribute so should be able to use one fragment for both?